### PR TITLE
Implement deficit payment mechanism + tests

### PR DESCRIPTION
**Pull request by Fletch153**

### CODE REVIEW

File: contracts/gas-snapshots/.gas-snapshot [hunk 1]
```diff
@@ -0,0 +1,285 @@
+AutomationForwarder_forward:testBasicSuccess() (gas: 87621)
```
Comment (RensR) [inline contracts/gas-snapshots/.gas-snapshot:1]: please remove and run `make snapshot` with a FOUNDRY_PROFILE set

File: contracts/src/v0.8/llo-feeds/dev/FeeManager.sol [hunk 2]
```diff
@@ -200,8 +208,10 @@ contract FeeManager is IFeeManager, ConfirmedOwner, TypeAndVersionInterface {
 
         //check that the contract has enough LINK before paying the fee
         if (reward.amount > IERC20(i_linkAddress).balanceOf(address(this))) {
-          // If not enough LINK on this contract to forward for rewards, fire this event and
-          // call onFeePaid out-of-band to pay out rewards
+          // If not enough LINK on this contract to forward for rewards, tally the deficit to be paid by out-of-band LINK
+          unchecked {
```
Comment (RensR) [inline contracts/src/v0.8/llo-feeds/dev/FeeManager.sol:212]: Do we really need unchecked here?

Comment (austinborn) [reply contracts/src/v0.8/llo-feeds/dev/FeeManager.sol]: Primary intention is gas savings since it's an easy change.

**Review by austinborn — DISMISSED** [review]

File: contracts/src/v0.8/llo-feeds/test/fee-manager/BaseFeeManager.t.sol [hunk 3]
```diff
@@ -341,4 +342,20 @@ contract BaseFeeManagerTest is Test {
...
+  function payLinkDeficit(bytes32 configDigest, address sender) public {
+    //record the current address and switch to the recipient
+    address originalAddr = msg.sender;
+    changePrank(sender);
+
+    //approve the link to be transferred
+    feeManager.payLinkDeficit(configDigest);
+
+    //change back to the original address
+    changePrank(originalAddr);
```
Comment (RensR) [inline contracts/src/v0.8/llo-feeds/test/fee-manager/BaseFeeManager.t.sol:355]: You don't have to change back the prank

File: contracts/src/v0.8/llo-feeds/test/fee-manager/BaseFeeManager.t.sol [hunk 4]
```diff
@@ -341,4 +342,20 @@ contract BaseFeeManagerTest is Test {
...
     return address(native);
   }
+
+  function payLinkDeficit(bytes32 configDigest, address sender) public {
+    //record the current address and switch to the recipient
+    address originalAddr = msg.sender;
+    changePrank(sender);
+
+    //approve the link to be transferred
+    feeManager.payLinkDeficit(configDigest);
```
Comment (RensR) [inline contracts/src/v0.8/llo-feeds/test/fee-manager/BaseFeeManager.t.sol:352]: missing expect emit, this is just testing that is doesn't revert

File: contracts/src/v0.8/llo-feeds/dev/FeeManager.sol [hunk 5]
```diff
@@ -350,7 +359,17 @@ contract FeeManager is IFeeManager, ConfirmedOwner, TypeAndVersionInterface {
...
+  function _getReportVersion(bytes32 feedId) internal pure returns (bytes32) {
     return REPORT_VERSION_MASK & feedId;
   }
+
+  /// @inheritdoc IFeeManager
+  function payLinkDeficit(bytes32 configDigest) external onlyOwner {
+    uint256 deficit = s_linkDeficit[configDigest];
+
+    delete s_linkDeficit[configDigest];
+    i_rewardManager.onFeePaid(configDigest, address(this), deficit);
```
Comment (RensR) [inline contracts/src/v0.8/llo-feeds/dev/FeeManager.sol:371]: Do you need a check on the current balance before doing this? onFeePaid seems to use transferFrom and not safeTransferFrom, I'd also recommend changing that.

Comment (Fletch153) [reply contracts/src/v0.8/llo-feeds/dev/FeeManager.sol:371]: They ultimately do the same thing for ERC20 tokens except safeTransferFrom uses more gas. transferFrom() will revert if the transfer fails.

**Review by austinborn — APPROVED** [review]

File: contracts/src/v0.8/llo-feeds/dev/FeeManager.sol (hunk 5 above)
Comment (Fletch153) [reply contracts/src/v0.8/llo-feeds/dev/FeeManager.sol:375]: It's always linked or wrapped native in the current implementation

**Review by samsondav — APPROVED** [review]

**Pull request merged**